### PR TITLE
[DOCFIX] Update contributor/Building-Alluxio-From-Source Doc

### PR DESCRIPTION
**Pull request by xichen01**
### What changes are proposed in this pull request?

Update contributor/Building-Alluxio-From-Source Doc

### Why are the changes needed?

The Chinese contributor/Building-Alluxio-From-Source document is not updated with the latest changes, this PR synchronizes these updates

### CODE REVIEW

File: docs/cn/contributor/Building-Alluxio-From-Source.md [hunk 1]
```diff
@@ -7,48 +7,72 @@ priority: 0
...
-这部分内容的前提条件是你已安装Java JDK 8或以上、Maven 3.3.9或以上。
+- [Java JDK 8或以上](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
+- [Maven 3.3.](http://maven.apache.org/download.cgi)
+- [Git](https://git-scm.org/downloads)
 
 ### Checkout源码
 
 从Github上获取主分支并编译：
 
-```console
```
Comment (maobaolong) [inline docs/cn/contributor/Building-Alluxio-From-Source.md:22]: Why you change to use shell language other than console?

Comment (xichen01) [reply docs/cn/contributor/Building-Alluxio-From-Source.md:22]: `shell` has been changed to`console`

File: docs/cn/contributor/Building-Alluxio-From-Source.md [hunk 2]
```diff
@@ -125,3 +172,33 @@ $ mvn install -P<HADOOP_PROFILE> -Dhadoop.version=<HADOOP_VERSION> -DskipTests
 -Phadoop-2 -Dhadoop.version=2.2.0.2.1.0.0-92
 -Phadoop-2 -Dhadoop.version=2.4.0.2.1.3.0-563
 ```
+
+{% endcollapsible %}
+{% endaccordion %}
+
+## 故障排除
+
+### java.lang.OutOfMemoryError的异常。Java堆空间
```
Comment (dbw9580) [inline docs/cn/contributor/Building-Alluxio-From-Source.md:181]:
```suggestion
### 编译时出现 OutOfMemoryError 的异常
```

File: docs/cn/contributor/Building-Alluxio-From-Source.md [hunk 3]
```diff
@@ -125,3 +172,33 @@ $ mvn install -P<HADOOP_PROFILE> -Dhadoop.version=<HADOOP_VERSION> -DskipTests
...
+
+### NullPointerException occurred while execute org.codehaus.mojo:buildnumber-maven-plugin:1.4:create
+
+如果你看到maven有以下错误信息。
+"`Failed to execute goal org.codehaus.mojo:buildnumber-maven-plugin:1.4:create-metadata (default) on project alluxio-core-common: Execution default of goal org.codehaus.mojo:buildnumber-maven-plugin:1.4:create-metadata failed: NullPointerException`"
+
+是由于`build number`是基于从SCM检索的`revision number`，它将从git哈希代码中检查`build number`。如果检查失败，SCM就会抛出NPE。为避免该异常，请用maven参数"`-Dmaven.buildNumber.revisionOnScmFailure`"来设置Alluxio版本。
+例如，如果alluxio的版本是2.7.3，那么请设置参数"`-Dmaven.buildNumber.revisionOnScmFailure=2.7.3`"。
+
+更多信息可见https://www.mojohaus.org/buildnumber-maven-plugin/create-mojo.html#revisionOnScmFailure。
```
Comment (dbw9580) [inline docs/cn/contributor/Building-Alluxio-From-Source.md:204]:
```suggestion
更多信息可见[此处](https://www.mojohaus.org/buildnumber-maven-plugin/create-mojo.html#revisionOnScmFailure)。
```

**Review by dbw9580 — CHANGES_REQUESTED** [review]
Thanks! A few minor issues.

File: docs/cn/contributor/Building-Alluxio-From-Source.md (hunk 3 above)
Comment (xichen01) [reply docs/cn/contributor/Building-Alluxio-From-Source.md]: done

File: docs/cn/contributor/Building-Alluxio-From-Source.md [hunk 4]
```diff
@@ -57,40 +81,60 @@ Maven编译环境将自动获取依赖，编译源码，运行单元测试，并
...
+
+例如,
+```shell
+$ mvn clean install -pl underfs/hdfs/ \
+  -Dmaven.javadoc.skip=true -DskipTests -Dlicense.skip=true \
+  -Dcheckstyle.skip=true -Dfindbugs.skip=true \
+  -Pufs-hadoop-3 -Dufs.hadoop.version=3.3.1
+```
+要启用`active sync`，请确保使用 `hdfsActiveSync` 属性来构建，
+请参考 [Active Sync for HDFS]({{ '/en/core-services/Unified-Namespace.html#active-sync-for-hdfs' | relativize_url }}) 获得更多关于使用`active sync`的信息。
```
[the remark this answers is not in the document]
Comment (xichen01) [reply docs/cn/contributor/Building-Alluxio-From-Source.md]: done

File: docs/cn/contributor/Building-Alluxio-From-Source.md (hunk 2 above)
Comment (xichen01) [reply docs/cn/contributor/Building-Alluxio-From-Source.md]: done

File: docs/cn/contributor/Building-Alluxio-From-Source.md [hunk 5]
```diff
@@ -7,48 +7,72 @@ priority: 0
...
 
-## 编译Alluxio
+该指南介绍如何克隆Alluxio仓库, 编译Alluxio源码, 并且在你的环境中运行测试。
 
-该指南介绍如何从头编译Alluxio。
+## 软件版本
 
-这部分内容的前提条件是你已安装Java JDK 8或以上、Maven 3.3.9或以上。
+- [Java JDK 8或以上](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html)
+- [Maven 3.3.](http://maven.apache.org/download.cgi)
```
[the remark this answers is not in the document]
Comment (xichen01) [reply docs/cn/contributor/Building-Alluxio-From-Source.md]: done

File: docs/cn/contributor/Building-Alluxio-From-Source.md [hunk 6]
```diff
@@ -7,48 +7,72 @@ priority: 0
...
+- [Git](https://git-scm.org/downloads)
 
 ### Checkout源码
 
 从Github上获取主分支并编译：
 
-```console
+```shell
```
[the remark this answers is not in the document]
Comment (xichen01) [reply docs/cn/contributor/Building-Alluxio-From-Source.md]: done

File: docs/cn/contributor/Building-Alluxio-From-Source.md [hunk 7]
```diff
@@ -7,48 +7,72 @@ priority: 0
...
 $ git tag
 $ git checkout <TAG_NAME>
 ```
 
+## (Optional) Checkout Building Environment Using Docker
+
+本节将知道您如何根据我们发布的docker镜像来构建一个编译环境。
```
[the remark this answers is not in the document]
Comment (xichen01) [reply docs/cn/contributor/Building-Alluxio-From-Source.md]: done

File: docs/cn/contributor/Building-Alluxio-From-Source.md [hunk 8]
```diff
@@ -57,40 +81,60 @@ Maven编译环境将自动获取依赖，编译源码，运行单元测试，并
...
-```console
+```shell
 $ ./bin/alluxio-stop.sh local
 ```
 
 ## 编译选项
 
 ### 计算框架支持
-自Alluxio 1.7开始，**你不要在Maven编译的时候使用不同计算框架的配置文件**。编译后位于`{{site.ALLUXIO_CLIENT_JAR_PATH}}`的Alluxio客户端jar包将适用于不同的计算框架（如：Spark、Flink，Presto等）。
+自Alluxio 1.7开始。编译后位于`{{site.ALLUXIO_CLIENT_JAR_PATH}}`的Alluxio客户端jar包将适用于不同的计算框架（如：Spark、Flink，Presto等）。
```
[the remark this answers is not in the document]
Comment (xichen01) [reply docs/cn/contributor/Building-Alluxio-From-Source.md]: done

File: docs/cn/contributor/Building-Alluxio-From-Source.md [hunk 9]
```diff
@@ -80,17 +104,37 @@ $ ./bin/alluxio-stop.sh local
...
+
+例如
+```console
+$ mvn clean install -pl underfs/hdfs/ \
+  -Dmaven.javadoc.skip=true -DskipTests -Dlicense.skip=true \
+  -Dcheckstyle.skip=true -Dfindbugs.skip=true \
+  -Pufs-hadoop-3 -Dufs.hadoop.version=3.3.1
+```
+要启用`active sync`，请确保使用 `hdfsActiveSync` 属性来构建，
+请参考 [Active Sync for HDFS]({{ '/cn/core-services/Unified-Namespace.html' | relativize_url }}#active-sync-for-hdfs) 获得更多关于使用Active Sync的信息。
```
Comment (dbw9580) [inline docs/cn/contributor/Building-Alluxio-From-Source.md:131]:
```suggestion
请参考 [Active Sync for HDFS]({{ '/cn/core-services/Unified-Namespace.html' | relativize_url }}#hdfs元数据主动同步) 获得更多关于使用Active Sync的信息。
```

File: docs/cn/contributor/Building-Alluxio-From-Source.md [hunk 10]
```diff
@@ -22,33 +24,55 @@ priority: 0
 ```console
 $ git clone git://github.com/alluxio/alluxio.git
 $ cd alluxio
+$ export ALLUXIO_HOME=$(pwd)
 ```
-您可以编译特定版本的Alluxio，否则这将编译源码的master分支。
+您可以编译特定版本的Alluxio，否则这将编译源码的master分支。如果你要构建特的版本的代码，你可以使用 git tag 获取。
```
Comment (dbw9580) [inline docs/cn/contributor/Building-Alluxio-From-Source.md:29]:
```suggestion
您可以编译特定版本的Alluxio，否则这将编译源码的master分支。如果你要构建特定版本的代码，你可以使用 git tag 获取。
```

**Review by dbw9580 — APPROVED** [review]
LGTM

**Review by maobaolong — APPROVED** [review]
LGTM

File: docs/cn/contributor/Building-Alluxio-From-Source.md (hunk 10 above)
Comment (xichen01) [reply docs/cn/contributor/Building-Alluxio-From-Source.md]: done'

File: docs/cn/contributor/Building-Alluxio-From-Source.md (hunk 9 above)
Comment (xichen01) [reply docs/cn/contributor/Building-Alluxio-From-Source.md]: done

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**